rust/stackablectl/CHANGELOG.md

```diff
@@ -4,11 +4,17 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+### Added
+
+- Pass the namespace as argument `NAMESPACE` to templated manifests.
+  This should unblock demos to run in all namespaces, as they can template the namespace e.g. for the fqdn of services ([#355]).
```

Comment:
suggestion: Adjust templating variable name

```suggestion
- Pass the product namespace as a templating variable `productNamespace` to manifests.
  This should unblock demos to run in all namespaces, as they can template the namespace e.g. for the FQDN of services ([#355]).
```

Reply:
Yeah, I see points for productNamespace vs NAMESPACE.

productNamespace:
* Similar to stack arguments
* More aligned with how the CLI flag is called

NAMESPACE:
* Similar to our kuttl tests
* The different casing makes it clear that this is a special variable, and not coming from stack.yaml (other than the others)
* One could argue it's needless to prefix "product" namespace, as there is no other potential namespace here.

Reply:
Picked the stylistic changes into 4905016

Reply:
I'm still in favor of keeping it identical to the CLI argument. There might come a time where we also want the operator namespace to be available as a templating variable, and I would like make it very clear what namespace is being talked about in the template.

Reply:
Ok, can we agree on `PRODUCT_NAMESPACE` than?

Reply:
I disgree with using `PRODUCT_NAMESPACE`. Products are one thing, but the demos and stacks are more than the SDP products.

I also think `stackablectl` should use `--operator-namespace` and `--namespace` (well known), rather than `--operator-namespace` and `--product-namespace` (bespoke, and less applicable when it comes to demos/stacks).

Reply:
I have to admin I'm with Nick here

Reply:
I don't have strong feelings for `PRODUCT_NAMESPACE` vs `NAMESPACE`. I'm more worried about consistency between the templating variable and the CLI argument. They must be the same.

I'm fine with getting this PR in and changing the CLI arguments in a follow-up PR. I just want to avoid us forgetting to change it.

I will only agree to a new release once the naming is the same in both places.

Reply:
I agree with the consistency, and don't mind if it is done in a follow up PR.
I would vote for removing `product` from the arg since it is not only about products.

ie:
- `--namespace` (defaults to `default`)
- `--operator-namespace` (defaults to `stackable-operators`)

:bulb: We could keep the old `--product-namespace` argument around for a release and emit a warning to use the `--namespace` arg instead.

Reply:
Thanks @NickLarsenNZ! The PR looks good to me

Reply:
Updated this PR, please have another look :)